src/clj_http/client.clj

```diff
@@ -1168,12 +1168,26 @@
      (throw (IllegalArgumentException. "Host URL cannot be nil"))))
 
 (defn- request*
-  [req [respond raise]]
-  (if (opt req :async)
+  [req & [respond raise]]
+  {:pre [(not (and (:async req) (:async-future req)))]}
```

Comment:
These should use `(opt req :async)` and `(opt req :async-future)` so they support the optional `?` suffix